lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -273,38 +271,41 @@ public function executeInserts()
         $stmt       = $this->conn->prepare($this->getInsertSQL());
         $tableName  = $this->class->getTableName();
 
-        foreach ($this->queuedInserts as $entity) {
-            $insertData = $this->prepareInsertData($entity);
+        foreach (array_chunk($this->queuedInserts, $this->getMaxBatchedInsertQueries()) as $queuedInsertsChunk) {
```

Comment:
As soon as you do that, you now have the contents of `$this->queuedInserts` twice in memory. Wouldn't it be better to extract the first `$this->getMaxBatchedInsertQueries()` elements out of that array with `array_slice()` until it is empty? I see it will be emptied anyway after the loop, on line 308.

Reply:
I think this is ok, as values in php are not duplicated if not modified (aka copy on modify)

Reply:
Doesn't look ok to me: https://3v4l.org/1LYhg
`array_chunk` does not create a simple copy, it creates another array entirely

Reply:
This information should be set by a new annotation `@BatchSize(size = 100)` that is configured at the entity level.